source/slang/slang-lower-to-ir.cpp

```diff
@@ -4415,8 +4415,12 @@ struct ExprLoweringVisitorBase : public ExprVisitor<Derived, LoweredValInfo>
             }
             else if (auto aggTypeDeclRef = declRef.as<AggTypeDecl>())
             {
-                List<IRInst*> args;
+                if (aggTypeDeclRef.as<InterfaceDecl>())
+                {
+                    return LoweredValInfo::simple(getBuilder()->emitDefaultConstruct(irType));
+                }
```

Comment:
I have a question here.
Can we cast to `InterfaceDecl` before casting to `AggTypeDecl`?
Or does it have to check `AggTypeDecl` first for a reason?

In other words, can this be changed to the following?
```
            else if (aggTypeDeclRef.as<InterfaceDecl>())
            {
                return LoweredValInfo::simple(getBuilder()->emitDefaultConstruct(irType));
            }
            else if (auto aggTypeDeclRef = declRef.as<AggTypeDecl>())
            {
```

Reply:
You are right, this is definitely cleaner. changed.